src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkThreshold();
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+            size++;
+            return;
+        }
+        for ( ;node != null; node = node.next) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = new Node<>(key, value,null);
+                size++;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K,V> node = table[index];
+        for ( ;node != null; node = node.next) {
+            if (Objects.equals(key, node.key)) {
+                return node.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void checkThreshold() {
+        if (size > (int) (DEFAULT_LOAD_FACTOR * table.length)) {
+            resize();
+        }
+    }
+
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        table = new Node[oldTable.length * 2];
+        size = 0;
+        for (Node<K, V> node : oldTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private int getHashcode(K key) {
```

Comment:
you don't need this method and logic on 76th line it's enough to return key.hashCode()